src/git.ts

```diff
@@ -0,0 +1,159 @@
+import * as core from '@actions/core'
+import {ChildProcess, spawn} from 'child_process'
+import {Octokit} from '@octokit/rest'
+import {delimiter} from 'path'
+
+const gitForWindowsUsrBinPath = 'C:/Program Files/Git/usr/bin'
+
+/*
+ * It looks a bit ridiculous to use 48 workers on a build agent that has only
+ * a two-core CPU, yet manual testing revealed that 64 workers would be _even
+ * better_. But at 92 workers, resources are starved so much that the checkout
+ * is not only much faster, but also fails.
+ *
+ * Let's stick with 48, which should avoid running out of resources, but still
+ * is much faster than, say, using only 2 workers.
+ */
+const GIT_CONFIG_PARAMETERS = `'checkout.workers=56'`
+
+async function clone(
+  url: string,
+  destination: string,
+  verbose: number | boolean,
+  cloneExtraOptions: string[] = []
+): Promise<void> {
+  if (verbose) core.notice(`Cloning ${url} to ${destination}`)
+  const child = spawn(
+    'git.exe',
+    [
+      'clone',
+      '--depth=1',
+      '--single-branch',
+      '--branch=main',
+      ...cloneExtraOptions,
+      url,
+      destination
+    ],
+    {
+      env: {
+        GIT_CONFIG_PARAMETERS
+      },
+      stdio: [undefined, 'inherit', 'inherit']
+    }
+  )
+  return new Promise<void>((resolve, reject) => {
+    child.on('close', code => {
+      if (code === 0) {
+        resolve()
+      } else {
+        reject(new Error(`tar: exited with code ${code}`))
+      }
+    })
+  })
+}
+
+export async function getGitClient(
```

Comment:
Maybe we could call this `getArtifacts` instead? Or `getGitClientArtifacts`?

Reply:
I'm going with `getViaGit`, to keep in the spirit of the `get()` function in `downloader.ts` that we still use for the `minimal` artifact.